### ISSUE TITLE
Bogart fail :(

### ISSUE THREAD

**Issue by tnguyen90** [marbl/canu#1576]
Hello there

Similar to some other users [#1168](https://github.com/marbl/canu/issues/1168), or [#1141 ](https://github.com/marbl/canu/issues/1168), during the Bogart stage, we get an error. We used the initial command:
```
canu -p TRL -d TRL_genome_pacbio genomeSize=3.2m -pacbio-raw lobster.fastq.gz
```
I'm aware that I might need to upload some files to the ftp server to fix this issue but I will ask first and willing to upload the files needed later. 

Here's the log code

```
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'oea' concurrent execution on Mon Dec  2 18:11:57 2019 with 8036.859 GB free disk space (6 processes; 63 concurrently)

    cd unitigging/3-overlapErrorAdjustment
    ./oea.sh 1 > ./oea.000001.out 2>&1
    ./oea.sh 2 > ./oea.000002.out 2>&1
    ./oea.sh 3 > ./oea.000003.out 2>&1
    ./oea.sh 4 > ./oea.000004.out 2>&1
    ./oea.sh 5 > ./oea.000005.out 2>&1
    ./oea.sh 6 > ./oea.000006.out 2>&1

-- Finished on Mon Dec  2 19:57:41 2019 (6345 seconds) with 8035.121 GB free disk space
----------------------------------------
-- Found 6 overlap error adjustment output files.
-- No change in report.
-- Finished stage 'overlapErrorAdjustmentCheck', reset canuIteration.
----------------------------------------
-- Starting command on Mon Dec  2 19:57:41 2019 with 8035.121 GB free disk space

    cd unitigging/3-overlapErrorAdjustment
    /home/tnguyen/anaconda3/envs/pacbio-py2.7/bin/loadErates \
      -S ../../TRL.seqStore \
      -O ../TRL.ovlStore \
      -L ./oea.files \
    > ./oea.apply.err 2>&1

-- Finished on Mon Dec  2 19:57:45 2019 (4 seconds) with 8033.383 GB free disk space
----------------------------------------
-- No report available.
-- Report changed.
-- Finished stage 'updateOverlapStore', reset canuIteration.
-- Finished stage 'unitig', reset canuIteration.
-- No change in report.
-- No change in report.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'bat' concurrent execution on Mon Dec  2 19:57:45 2019 with 8033.383 GB free disk space (1 processes; 1 concurrently)

    cd unitigging/4-unitigger
    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1

-- Finished on Mon Dec  2 20:10:25 2019 (760 seconds) with 8033.35 GB free disk space
----------------------------------------
--
-- Bogart failed, retry
--
-- No change in report.
--
-- Running jobs.  Second attempt out of 2.
----------------------------------------
-- Starting 'bat' concurrent execution on Mon Dec  2 20:10:25 2019 with 8033.35 GB free disk space (1 processes; 1 concurrently)

    cd unitigging/4-unitigger
    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1

-- Finished on Mon Dec  2 20:22:20 2019 (715 seconds) with 8033.35 GB free disk space
----------------------------------------
--
-- Bogart failed, tried 2 times, giving up.
--

ABORT:
ABORT: Canu snapshot ()
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT: Disk space available:  8033.35 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (unitigging/4-unitigger/unitigger.err):
ABORT:
ABORT:   optimizePositions()--     converged: 181294 reads
ABORT:   optimizePositions()--     changed:   837264 reads
ABORT:   optimizePositions()--   Recomputing positions, iteration 4, with 4 threads.
ABORT:   optimizePositions()--     Reset zero.
ABORT:   optimizePositions()--     Checking convergence.
ABORT:   optimizePositions()--     converged: 181356 reads
ABORT:   optimizePositions()--     changed:   837202 reads
ABORT:   optimizePositions()--   Recomputing positions, iteration 5, with 4 threads.
ABORT:   optimizePositions()--     Reset zero.
ABORT:   optimizePositions()--     Checking convergence.
ABORT:   optimizePositions()--     converged: 181391 reads
ABORT:   optimizePositions()--     changed:   837167 reads
ABORT:   optimizePositions()--   Expanding short reads with 4 threads.
ABORT:   optimizePositions()--   Updating positions.
ABORT:   optimizePositions()--   Finished.
ABORT:   splitDiscontinuous()-- Tested 1469 tigs, split 1 tig into 2 new tigs.
ABORT:   detectSpur() done.
ABORT:   tested           21
ABORT:   nEdges      5'   11   3'   11
ABORT:   nPotential        0         2
ABORT:   nVerified         0         0
ABORT:   
ABORT:   ==> PLACE CONTAINED READS.
ABORT:   
ABORT:   computeErrorProfiles()-- Computing error profiles for 14785 tigs, with 4 threads.
ABORT:   computeErrorProfiles()-- Finished.
ABORT:   
ABORT:   placeContains()-- placing 159166 contained and 852556 unplaced reads, with 4 threads.
ABORT:   placeContains()-- Placed 98743 contained reads and 227 unplaced reads.
ABORT:   placeContains()-- Failed to place 60423 contained reads (too high error suspected) and 852329 unplaced reads (lack of overlaps suspected).
ABORT:   optimizePositions()-- Optimizing read positions for 1018558 reads in 14785 tigs, with 4 threads.
ABORT:   optimizePositions()--   Allocating scratch space for 1018558 reads (63659 KB).
ABORT:   optimizePositions()--   Initializing positions with 4 threads.
ABORT:   bogart: bogart/AS_BAT_OptimizePositions.C:261: void Unitig::optimize_initPlace(uint32, optPos*, optPos*, bool, std::set<unsigned int>&, bool): Assertion `cnt > 0' failed.
ABORT:   
ABORT:   Failed with 'Aborted'; backtrace (libbacktrace):
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   /home/nwani/m3/conda-bld/compilers_linux-64_1560109574129/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/libgomp/parallel.c::171 in GOMP_parallel()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   ../sysdeps/x86_64/elf/start.S::103 in (null)()
ABORT:   Aborted (core dumped)
ABORT:
```

**Comment by tnguyen90** [comment]
Forgot to mention but it's Canu 1.9 installed with conda install canu

**Comment by tnguyen90** [comment]
Did a bit of investigation, here is the `unitigger.err` file

```
==> PARAMETERS.

Resources:
  Memory                16 GB
  Compute Threads       4 (command line)

Lengths:
  Minimum read          0 bases
  Minimum overlap       500 bases

Overlap Error Rates:
  Graph                 0.045 (4.500%)
  Max                   0.045 (4.500%)

Deviations:
  Graph                 12.000
  Bubble                12.000
  Repeat                6.000

Edge Confusion:
  Absolute              2100
  Percent               200.0000

Unitig Construction:
  Minimum intersection  500 bases
  Maxiumum placements   2 positions

Debugging Enabled:
  (none)

==> LOADING AND FILTERING OVERLAPS.

ReadInfo()-- Using 1018557 reads, no minimum read length used.

OverlapCache()-- limited to 16384MB memory (user supplied).

OverlapCache()--       7MB for read data.
OverlapCache()--      38MB for best edges.
OverlapCache()--     101MB for tigs.
OverlapCache()--      27MB for tigs - read layouts.
OverlapCache()--      38MB for tigs - error profiles.
OverlapCache()--    4096MB for tigs - error profile overlaps.
OverlapCache()--       0MB for other processes.
OverlapCache()-- ---------
OverlapCache()--    4329MB for data structures (sum of above).
OverlapCache()-- ---------
OverlapCache()--      19MB for overlap store structure.
OverlapCache()--   12035MB for overlap data.
OverlapCache()-- ---------
OverlapCache()--   16384MB allowed.
OverlapCache()--
OverlapCache()-- Retain at least 627 overlaps/read, based on 313.93x coverage.
OverlapCache()-- Initial guess at 774 overlaps/read.
OverlapCache()--
OverlapCache()-- Adjusting for sparse overlaps.
OverlapCache()--
OverlapCache()--               reads loading olaps          olaps               memory
OverlapCache()--   olaps/read       all      some          loaded                 free
OverlapCache()--   ----------   -------   -------     ----------- -------     --------
OverlapCache()--          774    947476     71081        65145315   6.98%      11041 MB
OverlapCache()--        10954    980151     38406       568812813  60.97%       3356 MB
OverlapCache()--        16680    991648     26909       756674799  81.11%        489 MB
OverlapCache()--        17872    994292     24265       787179437  84.38%         24 MB
OverlapCache()--        17936    994434     24123       788727655  84.54%          0 MB
OverlapCache()--        17937    994437     24120       788751778  84.54%          0 MB
OverlapCache()--
OverlapCache()-- Loading overlaps.
OverlapCache()--
OverlapCache()--          read from store           saved in cache
OverlapCache()--   ------------ ---------   ------------ ---------
OverlapCache()--      101748567 (010.91%)       83774974 (008.98%)
OverlapCache()--      205166460 (021.99%)      169018700 (018.12%)
OverlapCache()--      306545173 (032.86%)      252786788 (027.10%)
OverlapCache()--      404700322 (043.38%)      333434419 (035.74%)
OverlapCache()--      505425328 (054.18%)      416487681 (044.64%)
OverlapCache()--      610356618 (065.42%)      502559676 (053.87%)
OverlapCache()--      689533847 (073.91%)      568121921 (060.90%)
OverlapCache()--      761552091 (081.63%)      628038738 (067.32%)
OverlapCache()--      831217246 (089.10%)      686017388 (073.53%)
OverlapCache()--      916193925 (098.20%)      756255480 (081.06%)
OverlapCache()--   ------------ ---------   ------------ ---------
OverlapCache()--      932947112 (100.00%)      770309806 (082.57%)
OverlapCache()--
OverlapCache()-- Ignored 12067590 duplicate overlaps.
OverlapCache()--
OverlapCache()-- Symmetrizing overlaps.
OverlapCache()--   Finding missing twins.
OverlapCache()--   Found 118146114 missing twins in 770309806 overlaps, 303963 are strong.
OverlapCache()--   Dropping weak non-twin overlaps; allocated 0 MB scratch space.
OverlapCache()--   Dropped 25948118 overlaps; scratch space released.
OverlapCache()--   Adding 92197996 missing twin overlaps.
OverlapCache()--   Finished.

BestOverlapGraph()-- allocating best edges (31MB)

BestOverlapGraph()-- finding initial best edges.

BestOverlapGraph()-- filtering suspicious reads.
BestOverlapGraph()-- marked 837877 reads as suspicious.

BestOverlapGraph()-- filtering high error edges.

BestOverlapGraph()-- filtering reads with lopsided best edges.

BestOverlapGraph()-- filtering spur reads.
BestOverlapGraph()-- detected 11291 spur reads and 841339 singleton reads.
BestOverlapGraph()-- detected 3049 zombie reads.

BestOverlapGraph()-- removing best edges for contained reads.

==> BUILDING GREEDY TIGS.

breakSingletonTigs()-- Removed 10953 singleton tigs; reads are now unplaced.
optimizePositions()-- Optimizing read positions for 1018558 reads in 14783 tigs, with 4 threads.
optimizePositions()--   Allocating scratch space for 1018558 reads (63659 KB).
optimizePositions()--   Initializing positions with 4 threads.
optimizePositions()--   Recomputing positions, iteration 1, with 4 threads.
optimizePositions()--     Reset zero.
optimizePositions()--     Checking convergence.
optimizePositions()--     converged: 180705 reads
optimizePositions()--     changed:   837853 reads
optimizePositions()--   Recomputing positions, iteration 2, with 4 threads.
optimizePositions()--     Reset zero.
optimizePositions()--     Checking convergence.
optimizePositions()--     converged: 181141 reads
optimizePositions()--     changed:   837417 reads
optimizePositions()--   Recomputing positions, iteration 3, with 4 threads.
optimizePositions()--     Reset zero.
optimizePositions()--     Checking convergence.
optimizePositions()--     converged: 181294 reads
optimizePositions()--     changed:   837264 reads
optimizePositions()--   Recomputing positions, iteration 4, with 4 threads.
optimizePositions()--     Reset zero.
optimizePositions()--     Checking convergence.
optimizePositions()--     converged: 181356 reads
optimizePositions()--     changed:   837202 reads
optimizePositions()--   Recomputing positions, iteration 5, with 4 threads.
optimizePositions()--     Reset zero.
optimizePositions()--     Checking convergence.
optimizePositions()--     converged: 181391 reads
optimizePositions()--     changed:   837167 reads
optimizePositions()--   Expanding short reads with 4 threads.
optimizePositions()--   Updating positions.
optimizePositions()--   Finished.
splitDiscontinuous()-- Tested 1469 tigs, split 1 tig into 2 new tigs.
detectSpur() done.
tested           21
nEdges      5'   11   3'   11
nPotential        0         2
nVerified         0         0

==> PLACE CONTAINED READS.

computeErrorProfiles()-- Computing error profiles for 14785 tigs, with 4 threads.
computeErrorProfiles()-- Finished.

placeContains()-- placing 159166 contained and 852556 unplaced reads, with 4 threads.
placeContains()-- Placed 98743 contained reads and 227 unplaced reads.
placeContains()-- Failed to place 60423 contained reads (too high error suspected) and 852329 unplaced reads (lack of overlaps suspected).
optimizePositions()-- Optimizing read positions for 1018558 reads in 14785 tigs, with 4 threads.
optimizePositions()--   Allocating scratch space for 1018558 reads (63659 KB).
optimizePositions()--   Initializing positions with 4 threads.
bogart: bogart/AS_BAT_OptimizePositions.C:261: void Unitig::optimize_initPlace(uint32, optPos*, optPos*, bool, std::set<unsigned int>&, bool): Assertion `cnt > 0' failed.

Failed with 'Aborted'; backtrace (libbacktrace):
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
/home/nwani/m3/conda-bld/compilers_linux-64_1560109574129/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/libgomp/team.c::123 in gomp_thread_start()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
Aborted (core dumped)`
```

**Comment by skoren** [comment]
How, when was Canu installed? I ask because it's reporting the version as "snapshot" and there was briefly a bug in this code that was fixed before the final 1.9 release (see #1520). 

If you're using the final release, are you able to share the data? The seqStore and unitigging/ovlStore should be enough to test/reproduce locally.

**Comment by tnguyen90** [comment]
awww, I think I fixed it, somehow conda install 1.8 instead of 1.9. The assembly finalized for me. Thanks !
